# BIO613: DNA Sequence Analysis Project


Author
**Student: Group (Git & Github)**


**Course: BIO-613 Programming for Bioinformatics**



**Institution: National University-Sudan**



**Date: September 2026**

In [1]:
from Bio.Seq import Seq

dna = Seq("ATGGCGCGCTTA")
gc_content = (dna.count("G") + dna.count("C")) / len(dna) * 100
print("DNA Sequence:", dna)
print(f"GC Content: {round(gc_content, 2)}%")

DNA Sequence: ATGGCGCGCTTA
GC Content: 58.33%


In [2]:
from Bio.Seq import Seq

dna = Seq("ATGGCGCGCTTA")
at_content = (dna.count("A") + dna.count("T")) / len(dna) * 100
print("DNA Sequence:", dna)
print(f"AT Content: {round(at_content, 2)}%")

DNA Sequence: ATGGCGCGCTTA
AT Content: 41.67%


In [3]:
from Bio.Seq import Seq

dna = Seq("ATGGCGCGCTTA")

# Length
seq_length = len(dna)

# Reverse Complement
rev_comp = dna.reverse_complement()

# Translation
protein_seq = dna.translate()

print(f"Sequence Length: {seq_length} bp")
print(f"Reverse Complement: {rev_comp}")
print(f"Translated Protein: {protein_seq}")

Sequence Length: 12 bp
Reverse Complement: TAAGCGCGCCAT
Translated Protein: MARL


## Conclusion & Discussion

In this project, a comprehensive computational sequence analysis was successfully performed on the target DNA sequence using **BioPython**:

1. **Nucleotide Composition Analysis:**
   * **GC Content:** The calculated GC ratio is **58.33%**, indicating moderate-to-high thermal stability and strong double-stranded DNA binding due to the higher proportion of G-C triple hydrogen bonds.
   * **AT Content:** The calculated AT ratio is **41.67%**, complementing the sequence composition profile.

2. **Sequence Characterization & Translation:**
   * **Sequence Length:** The sequence comprises **12 base pairs (bp)**.
   * **Reverse Complement:** The reverse complementary strand was generated as `TAAGCGCGCCAT`, representing the antiparallel strand required for molecular base pairing.
   * **Protein Translation:** Translation of the nucleotide sequence yielded the peptide sequence **`MARL`** (Methionine-Alanine-Arginine-Leucine), demonstrating the open reading frame's functional amino acid output.

Overall, the pipeline effectively demonstrates the utility of automated Python scripts for rapid genomic sequence profiling and functional annotation.


